controllers/mirrorpeer_controller.go

```diff
@@ -550,14 +550,14 @@ func (r *MirrorPeerReconciler) createClusterRoleBindingsForSpoke(ctx context.Con
 		ObjectMeta: metav1.ObjectMeta{
 			Name: spokeClusterRoleBindingName,
 		},
-		Subjects: subjects,
-		RoleRef: rbacv1.RoleRef{
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, &spokeRoleBinding, func() error {
+		spokeRoleBinding.Subjects = subjects
+		spokeRoleBinding.RoleRef = rbacv1.RoleRef{
```

Comment:
RoleRef is immutable and we should not try to update it.https://github.com/red-hat-storage/ocs-operator/blob/799902be7f330ed2750b687205c849e2a66075bc/controllers/storagecluster/exporter.go#L634-L669 is one way to handle it.

Reply:
Added the check to update roleRef on new objects